libraries/network/src/main/kotlin/io/element/android/libraries/network/NetworkModule.kt

```diff
@@ -26,21 +27,34 @@ object NetworkModule {
     @Provides
     @SingleIn(AppScope::class)
     fun providesOkHttpClient(
-        buildMeta: BuildMeta,
         userAgentInterceptor: UserAgentInterceptor,
+        dynamicHttpLoggingInterceptor: DynamicHttpLoggingInterceptor,
     ): OkHttpClient = OkHttpClient.Builder().apply {
         connectTimeout(30, TimeUnit.SECONDS)
         readTimeout(60, TimeUnit.SECONDS)
         writeTimeout(60, TimeUnit.SECONDS)
         addInterceptor(userAgentInterceptor)
-        if (buildMeta.isDebuggable) addInterceptor(providesHttpLoggingInterceptor())
+        addInterceptor(dynamicHttpLoggingInterceptor)
     }.build()
-}
 
-private fun providesHttpLoggingInterceptor(): HttpLoggingInterceptor {
-    val loggingLevel = HttpLoggingInterceptor.Level.BODY
-    val logger = FormattedJsonHttpLogger(loggingLevel)
-    val interceptor = HttpLoggingInterceptor(logger)
-    interceptor.level = loggingLevel
-    return interceptor
+    @Provides
+    fun providesHttpLoggingInterceptor(): HttpLoggingInterceptor {
+        val loggingLevel = HttpLoggingInterceptor.Level.BODY
+        val logger = FormattedJsonHttpLogger(loggingLevel)
+        val interceptor = HttpLoggingInterceptor(logger)
+        interceptor.level = loggingLevel
+        return interceptor
+    }
+
+    @Provides
+    @SingleIn(AppScope::class)
+    fun provideDynamicHttpLoggingInterceptor(
```

Comment:
`@Provides` should be used only for classes that we do not own. In the case of `DynamicHttpLoggingInterceptor`, the constructor can just be annoted with `@Inject` (and `@SingleIn(AppScope::class)`)